### PR TITLE
Speedup HA core auth

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

The change is less as maybe it can look at first view. It's like the speed dial for wifi. Timing attacks with that are only possible if you know the password and it returns anyway True. The old password maybe still working sometimes but we speak about add-ons and the new password work all time.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes: https://github.com/home-assistant/hassio-addons/issues/1630
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/688

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by ludeeus** [comment]
Wouldn't this make auth be successful, even if the user object has changed in core?

**Comment by pvizeli** [comment]
Sounds worse at it is in context for what this function is used

**Comment by pvizeli** [comment]
I added an API for reset data with cli if user want force it

File: supervisor/api/__init__.py [hunk 1]
```diff
@@ -221,7 +221,11 @@ def _register_auth(self) -> None:
         api_auth.coresys = self.coresys
 
         self.webapp.add_routes(
-            [web.post("/auth", api_auth.auth), web.post("/auth/reset", api_auth.reset)]
+            [
+                web.post("/auth", api_auth.auth),
+                web.post("/auth/reset", api_auth.reset),
+                web.post("/auth/cache", api_auth.cache),
```
Comment (ludeeus) [inline supervisor/api/__init__.py]:
```suggestion
                web.delete("/auth/cache", api_auth.cache),
```

Should be delete

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: supervisor/api/__init__.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/api/__init__.py]: Lol... you already did that.

**Review by ludeeus — APPROVED** [review]

**Comment by balloob** [comment]
How often do we hit this API that it is worth potentially serving an outdated one from the cache instead of accessing the real source?

**Comment by pvizeli** [comment]
I.e. mqtt hit that a lot for every connection. Some devices just connects send state and disconnect.

**Comment by pvizeli** [comment]
Old:
Someone change the password (anyway not often happening on a service account). Everything stop impotently working.

Now:
Sone changes the password. All working until the cache is expired or someone logs in with new service accounts credentials.

The second provides anyway a better user experience and for pro users, they can run `ha auth cache` to force the old experience if they are really imminently needed.

**Comment by pvizeli** [comment]
@balloob I implemented an extended cache handling with background jobs. I think now it's like in old day but faster

File: supervisor/auth.py [hunk 2]
```diff
@@ -61,11 +68,29 @@ async def check_login(self, addon: Addon, username: str, password: str) -> bool:
...
             _LOGGER.debug("Home Assistant not running, checking cache")
-            return self._check_cache(username, password)
+            return cache_hit is True
+
+        # No cache hit
+        if cache_hit is None:
+            return await self._backend_login(addon, username, password)
+
+        # Home Assistant Core take over 1-2sec to validate it
+        # Let's use the cache and if they don't hit, we ask on core
```
Comment (ludeeus) [inline supervisor/auth.py:84]: This comment looks out of place

**Review by ludeeus — APPROVED** [review]

**Pull request merged**